File: pikashop-merchant-interface/src/main/java/com/chao/service/GoodsService.java
package com.chao.service;

import com.chao.common.viewobject.CommonResult;
import com.chao.entity.viewobject.GoodsVO;
import com.chao.mybatis.pojo.GoodsDo;

import java.util.List;

/**
 * 服务层接口
 * @author Administrator
 *
 */
public interface GoodsService {

	/**
	 * 返回全部列表
	 * @return
	 */
	public List<GoodsDo> findAll();
	
	
	/**
	 * 返回分页列表
	 * @return
	 */
	public CommonResult<GoodsDo> findPage(int pageNum, int pageSize);
	
	
	/**
	 * 增加
	*/
	public void add(GoodsVO goods);
	
	
	/**
	 * 修改
	 */
	public void update(GoodsVO goods);
	

	/**
	 * 根据ID获取实体
	 * @param id
	 * @return
	 */
	public GoodsDo findOne(Long id);

	public GoodsVO findOneDetail(Long id);
	
	/**
	 * 批量删除
	 * @param ids
	 */
	public void delete(Long[] ids);

	/**
	 * 分页
	 * @param pageNum 当前页 码
	 * @param pageSize 每页记录数
	 * @return
	 */
	public CommonResult<GoodsDo> findPage(GoodsDo goods, int pageNum, int pageSize);
	
}

File: pikashop-seller/src/main/webapp/js/service/goodsService.js
app.service('goodsService',function($http){
	//读取列表数据绑定到表单中
	this.findAll=function(){
		return $http.get('/biz/goods/list');
	}
	//分页 
	this.findPage=function(page,rows){
		return $http.get('/biz/goods/page?page='+page+'&rows='+rows);
	}
	//查询实体
	this.findOne=function(id){
		return $http.get('/biz/goods/one?id='+id);
	}
	this.findDetail=function(id){
		return $http.get('/biz/goods/detail?id='+id);
	}
	//增加 
	this.add=function(entity){
		return  $http.post('/biz/goods/insert',entity );
	}
	//修改 
	this.update=function(entity){
		return  $http.post('/biz/goods/update',entity );
	}
	//删除
	this.dele=function(ids){
		return $http.get('/biz/goods/delete?ids='+ids);
	}
	//搜索
	this.search=function(page,rows,searchEntity){
		return $http.post('/biz/goods/search?page='+page+"&rows="+rows, searchEntity);
	}    	
});
